### DISCUSSION TITLE
Question about GPU execution support

### DISCUSSION THREAD

**Discussion by Inglezos** [lisphilar/covid19-sir#497] [General]
I notice that even on Google Colab, the CovsirPhy codes run only on CPU, while if GPU is enabled, it runs again on CPU. Could we do something to parallelize the codes in order to be able to run on GPUs as well if the user wants so in order to reduce the runtime execution for many countries analysis (especially for simulation)? Is something like that possible at all in the first place?

**Comment by lisphilar** [comment]
It is diffucult to use GPU with the current version because scikit-learn (solve_ivp for simulation) and Optuna package (for suggestion of parameter sets) do not support GPU.
https://scikit-learn.org/stable/faq.html#will-you-add-gpu-support
https://github.com/optuna/optuna/issues/313

Alternatively, can we replace the parameter estimation codes with C language? I have little knowledge on C..., but this could be a solution.
